### PR TITLE
MAILBOX-337 Parrallelize JMAP message deletion

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: mailbox/store/src/test/java/org/apache/james/mailbox/store/AbstractMessageIdManagerStorageTest.java [hunk 1]
```diff
@@ -290,6 +302,16 @@ public void deleteMessageShouldRemoveMessageOnlyFromMailbox() throws Exception {
         assertThat(messageResults.get(0).getMailboxId()).isEqualTo(aliceMailbox2.getMailboxId());
     }
 
+    @Test
+    public void deleteAllShouldRemoveMessageFromAllMailbox() throws Exception {
```
Comment (rouazana) [inline mailbox/store/src/test/java/org/apache/james/mailbox/store/AbstractMessageIdManagerStorageTest.java]: s/Mailbox/Mailboxes

**Comment by aduprat** [comment]
```
Tests in error: 
  SpamAssassinContract.deletingSpamMessageShouldNotImpactSpamAssassinState:538 » ConditionTimeout
```
test this please

**Comment by aduprat** [comment]
```
Failed tests: 
  SpamAssassinInvokerTest.learnAsSpamShouldReturnTrueWhenLearningWorks:94 expected:<[tru]e> but was:<[fals]e>
```
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by mbaechler** [comment]
merged

**Pull request closed without merging**